release.go

```diff
@@ -248,11 +249,34 @@ func buildReleaseCommits(ctx context.Context, ghClient *githubClient, commits []
 			return nil, err
 		}
 		for i := range v {
+			sha := *v[i].MergeCommitSHA
+			shaPRs[sha] = v[i]
 			number := *v[i].Number
-			prs[number] = v[i]
+			numPRs[number] = v[i]
 		}
 	}
 
+	getPullRequest := func(commit Commit) (*github.PullRequest, error) {
+		if !commit.IsMerge() {
+			return nil, nil
+		}
+		if pr, ok := shaPRs[commit.Hash]; ok {
```

Comment:
This is not a bugfix, but an improvement.

The subject of a merge commit may not match `Merge pull request #([0-9]+) from .+`. So use merge commit SHA instead of PR Number to identify the PR.